src/doc/rustc-dev-guide/src/tests/ui.md

```diff
@@ -309,7 +309,8 @@ fn main((ؼ
 
 Use `//~?` to match an error without line information.
 `//~?` is precise and will not match errors if their line information is available.
-It should be preferred to using `error-pattern`, which is imprecise and non-exhaustive.
+For compile time tests, it should be preferred to using `error-pattern`, which
+is imprecise and non-exhaustive.
```

Comment:
Nit: this might be worded a bit confusingly, maybe

> For tests wishing to match against compiler diagnostics, error annotations should be preferred over `//@ error-pattern`, `//@ error-pattern` is imprecise and non-exhaustive.